Tools/scripts/uploader.py

```diff
@@ -786,6 +790,39 @@ def main():
                     # Go to the next port
                     continue
 
+                if args.identify:
+                    print(" Flash size: %u" % up.fw_maxsize)
+                    print(" board_type: %u" % up.board_type)
+                    print(" board_rev: %u" % up.board_rev)
+                    if up.bl_rev >= 5:
+                        chip = up.getCHIP()
+                        MCU_REV_STM32F4_REV_A = 0x1000
```

Comment:
this won't work for F7 and H7 chips

Reply:
Fixed.  Sort of.  We don't extract much information from the F7s etc.